src/runtime/analytics-service.js

```diff
@@ -122,9 +134,7 @@ export class AnalyticsService {
     if (source) {
       this.context_.setUtmSource(source);
     }
-    return this.xid_.get().then(id => {
-      this.context_.setTransactionId(id);
-    });
+    this.context_.setTransactionId(uuidFast());
```

Comment:
That will reset it in an expected way if it was already initialized. Let's do 

```
// Make sure the transactionId is initialized.
this.context_.setTransactionId(this.getTransactionId());
```

And this situation needs a test. Again, the repro scenario:
1. Call `const value = getTransactionId()`
2. Then get `this.setContext_()` to be called.
3. As the result, this should be true: `value == this.getTransactionId()`. Or in other words, transaction id has not changed.

Reply:
Done